common/types/activation.go

```diff
@@ -385,6 +386,14 @@ func (atx *ActivationTx) SetEffectiveNumUnits(numUnits uint32) {
 	atx.effectiveNumUnits = numUnits
 }
 
+func (atx *ActivationTx) SetReceived(received time.Time) {
+	atx.received = received
+}
+
+func (atx *ActivationTx) Received() time.Time {
```

Comment:
no. received time is determined by the receiving node, not the generating node. this is used to determine whether the receiving node will consider such ATX eligible in the node's view of active set, or in the beacon case, eligible for making beacon proposals.

Reply:
I understand but I mean that when we persist an ATX we should check if received is set (by the receiving node) to avoid storing an ATX with the time `January 1, year 1, 00:00:00.000000000 UTC`.

`Verify` would be imo be a good place to do that, but we could also do it in the `atx.Add` function.

Reply:
missed this earlier. done. thanks